pkg/apiserver/server.go

```diff
@@ -32,14 +37,23 @@ type ApiServer struct {
 	listener net.Listener
 }
 
-func New(definer Definer) kernel.ModuleFactory {
+func New(definer Definer, options ...Option) kernel.ModuleFactory {
 	return func(ctx context.Context, config cfg.Config, logger mon.Logger) (kernel.Module, error) {
-		settings := &Settings{
-			Port:         config.GetString("api_port"),
-			Mode:         config.GetString("api_mode"),
-			TimeoutRead:  config.GetDuration("api_timeout_read"),
-			TimeoutWrite: config.GetDuration("api_timeout_write"),
-			TimeoutIdle:  config.GetDuration("api_timeout_idle"),
+		var settings *Settings
+		if config.IsSet("api_port") {
```

Comment:
I really like the idea of modernizing the config structure, but do we want to invest the time to enforce the new structure?  
I have the feeling that we won't get rid of the old stuff otherwise.. 
To even push this one step further, i would like to rename the package from `apiserver` to `httpserver`.. What do you think?

Reply:
I think I would release this, giving the option to switch to the new structure. Then, in a later version I would drop the old structure and replace the check with a failure (i.e., if you still use the old format, the apiserver doesn't start so you see it isn't working anymore instead of silently have your settings ignored).
I really don't have a string opinion about the naming, I think `apiserver` is okay and sounds like quite some changes needed... So I wouldn't change it, but it wouldn't be too hard to actually change it and if you think this makes sense, I also wouldn't oppose that

Reply:
yeah, this is how you would do it with a library which is used a lot by external projects.. give them some time to migrate.. in our case i still like the "move fast" approach a lot and just do the hard cut.. than it's done and of the table..  
But i also don't want to force you to do it.. :grimacing: